### ISSUE TITLE
Multiple jobs (Rails mailers) scheduled at the same time

### ISSUE THREAD

**Issue by knightq** [sidekiq-cron/sidekiq-cron#400]
I've noticed this when I upgraded from 1.5.1 to 1.6.0 and above (tried up to 1.9.1, which is currently the last one).

We have 20 different instances running Sidekiq, with different configurations.
They all run with different config yml files that specifies distinct queues and for all them the Sidekiq `concurrency` has been set to `2`; here they are:
 
   - **Scheduler** worker service: 1 instance only: this is the only one sidekiq process that starts with `SCHEDULED_JOB_ENABLED=true`.
   - Worker service on queue set 1: up to 5 instances (autoscaling)
   - Worker service on queue set 2: up to 4 instances (autoscaling)
   - Worker service on queue set 3: up to 3 instances (autoscaling)
   - Worker service on queue set 4: up to 5 instances (autoscaling)
   - Worker service on queue set 5: 1 instance only
   - Worker service on queue set 6: 1 instance only

Even if I've read that `sidekiq-cron` support multiple instances, we have changed the initialiser so that it runs the scheduler only on the sigle Scheduled worker service (the first one of the list above ☝️)

```ruby  # sidekiq-cron configuration
# config/initializer/sidekiq.rb
Sidekiq.configure_server do |config|
  if ENV['SCHEDULED_JOB_ENABLED'].to_s.casecmp?('true')
    Rails.logger.info 'Cron Jobs are ENABLED!'
    config.on(:startup) do
      schedule_file = 'config/cron_schedule.yml'
      Sidekiq::Cron::Job.load_from_hash YAML.load_file(schedule_file) if File.exist?(schedule_file)
    end
  end
end
```
As you may have noticed, I've also tried to change the default file name to `cron_schedule.yml`, to manually handle the schedule load, as since [version 1.6.0](https://github.com/sidekiq-cron/sidekiq-cron/blob/master/CHANGELOG.md#160) it has been added the support for auto-loading the `config/schedule.yml` file and I though that was the problem (but it looks like it isn't, actually...).

Here is one example of scheduled job that gets enqueued **twice**:
```yaml
report_one:
  cron: 'every day at 8:30 UTC'
  class: MyApp::ReportOne
```

#### Environment
  - ruby 2.7.7
  - Sidekiq 7.0.5 / SidekiqPro 7.0.7
  - Redis 6.2.6 (on AWS ElasticCache cluster with single node)
  - sidekiq-cron 1.9.1 (but the problem started since 1.6.0)

I've already seen the other similar [issue#81](https://github.com/sidekiq-cron/sidekiq-cron/issues/81), but even if the problem is the same, the root cause is different.

Any ideas why this happens ?

**Comment by markets** [comment]
Hello @knightq 

Your setup seems seems fine 🤔 

Between those versions the most relevant change is the new autoloader: https://github.com/sidekiq-cron/sidekiq-cron/compare/v1.5.1...v1.6.0. So maybe you still have some "old" jobs enqueued? We've merged some autoloader related patches, like #405, maybe you can try with `master`?

Not sure if this is relevant in your case, but in case it helps: https://github.com/sidekiq-cron/sidekiq-cron/issues/158#issuecomment-296994313

**Comment by knightq** [comment]
Thank you @markets for the hint: really appreciated!
I already tried to exploit `sidekiq-uniq-job` `until_executed` lock, but it didn't work.
I'm going to try this further step (`until_and_while_executing`), which could catch this weird case.

I'll let you know whether it will work or not.

BTW, I also tried to lower the concurrency to 1 (from 2) to see if that was the problem, but it didn't solve either.
